demo/app/ios/Runner/flutterPlugin.swift

```diff
@@ -528,12 +531,32 @@ public class SwiftWalletSDKPlugin: NSObject, FlutterPlugin {
                                                                      vcCredentials: convertToVerifiableCredentialsArray(credentials: vcCredentials))
             result(displayDataResp)
           } catch let error as NSError {
+              let parsedError = WalleterrorParse(error.localizedDescription)
                return result(FlutterError.init(code: "Exception",
                                          message: "error while resolving credential",
-                                         details: error.localizedDescription))
+                                               details: parsedError))
+              
             }
     }
     
+    public func parseWalletSDKError(arguments: Dictionary<String, Any>, result: @escaping FlutterResult){
+        guard let localizedErrorMessage = arguments["localizedErrorMessage"] as? String else{
+            return  result(FlutterError.init(code: "NATIVE_ERR",
+                                             message: "error while parsing wallet sdk error",
+                                             details: "parameter localizedErrorMessage is missed"))
+        }
+        let parsedError = WalleterrorParse(localizedErrorMessage)
+        
+        var parsedErrorResult :[String:Any] = [
+            "category":   parsedError!.category,
+            "details":  parsedError!.details,
+            "code":  parsedError!.code,
+            "traceID":  parsedError!.traceID
+        ]
+        result(parsedErrorResult)
+    }
```

Comment:
Can you do the force unwrap once on line 548 instead of on every line from 551-554? Like this:

`let parsedError = WalleterrorParse(localizedErrorMessage)!`